### ISSUE TITLE
dart2js: specialize method bodies to arguments if known

### ISSUE THREAD

**Issue by hterkelsen** [dart-lang/sdk#24354]
consider this example program: (dead code added to the Foo constructor to make sure it isn't inlined)
```dart
void main() {
  new Foo(false);
}

class Foo {
  Foo(x) {
    if (x) {
      print('a');
    } else {
      print('b');
    }
    int i = 1;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
    i = i;
  }
}
```

The compiler should know that the only caller of the Foo constructor passes 'false' as the argument and specialize the body to eliminate the true branch of the if-statement.

**Comment by hterkelsen** [comment]
This is related to #24355 which together with this issue is causing dart2js not to remove unused code

**Comment by hterkelsen** [comment]
Closed with 28a803d3435c6565227ad31cb5db6cbe2173ca25
